rover_video/src/camera_node.cpp

```diff
@@ -0,0 +1,558 @@
+            return false;
+        }
+
+        // Read a single frame
+        cv::Mat frame;
+        bool ret = cap.read(frame);
+
+        if (ret)
+        {
+            // Save the frame as a sceenshot:
+            cv::imwrite(captureName, frame);
+            RCLCPP_INFO(LOGGER, "Screenshot saved successfully as: %s", captureName.c_str());
+
+            // Display the frame for debug
+            // cv::imshow("IP Camera Screenshot", frame);
+            // cv::waitKey(0);  // Wait for a key press
+            // cv::destroyAllWindows();
+        }
+        else
+        {
+            RCLCPP_ERROR(LOGGER, "Failed to capture frame from camera.");
+            return false;
+        }
+
+        // Release the video capture object
+        cap.release();
+
+        return true;
+    }
+}
+
+/**
+ * @brief Stops a recording that is being currently made
+ *
+ * @param cameraURL_ RTSP url of the camera that is currently being used
+ * @return true if successfully stopped the recording.
+ * @return false if unsuccesful
+ */
+bool CameraNode::stopRecording(std::string cameraURL_)
+{
+    std::lock_guard<std::mutex> lock(_recordingMutex);
+
+    if (_RecordingMap.find(cameraURL_) != _RecordingMap.end())
+    {
+        _RecordingMap.erase(cameraURL_);
+
+        if (_RecordingMap.empty())
+        {
+            _watchDogStop.store(true);
+            _recordingCv.notify_one();
+        }
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+/**
+ * @brief Creates a new recording inside the hashmap if it doesn't already exists.
+ *
+ * @param videoFolderPath_ Absolute path to the saving folder
+ * @param filename_ Name for the recording
+ * @param cameraURL_ RTSP url of the camera that is currently being used
+ * @return true
+ * @return false
+ */
+bool CameraNode::newRecording(std::string videoFolderPath_, std::string filename_, std::string cameraURL_)
+{
+    if (_RecordingMap.find(cameraURL_) != _RecordingMap.end())  // check if recording doesn't already exist
+    {
+        return false;
+    }
+    else
+    {
+        _RecordingMap.emplace(cameraURL_,
+                              Recording(videoFolderPath_,
+                                        filename_,
+                                        cameraURL_,
+                                        std::make_shared<rclcpp::Logger>(LOGGER),
+                                        [this](std::string url)
+                                        {
+                                            RequestShutdown(url);
+                                        }));
+
+        if (!_videoWatchDog.joinable())
+        {
+            StartWatchDog();
+        }
+
+        // Access the recording using at() to safely get the reference
+        Recording* pRecording = &_RecordingMap.at(cameraURL_);
+
+        if (pRecording->startRecording())
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
```

Comment:
Use returned reference instead pls 
```suggestion
        Recording& recording = _RecordingMap.at(cameraURL_);

        if (recording.startRecording())
        {
            return true;
        }
        else
        {
            return false;
        }
    }
```